### DISCUSSION TITLE
Troubleshooting SSL/TLS Error When Installing Cypress in Corporate Environment

### DISCUSSION THREAD

**Discussion by KSHCR7** [cypress-io/cypress#28512] [Questions and Help]
I am trying to install Cypress in my corporate environment using npm, but I am encountering an SSL/TLS error. 
I suspect it might be related to our corporate proxy settings 

**Error Message**: 
When I run `npm install cypress --save-dev`, I received the following error: 
`npm ERR! code ERR_SSL_TLSV1_UNRECOGNIZED_NAME npm ERR! errno ERR_SSL_TLSV1_UNRECOGNIZED_NAME npm ERR! request to https://proxy.example.net/cypress failed, reason: D04D0000:error:0A000458:SSL routines:ssl3_read_bytes:tlsv1 unrecognized name:c:\ws\deps\openssl\openssl\ssl\record\rec_layer_s3.c:1586:SSL alert number 112`

**Environment**: 
Operating System: Windows 
Node.js version: Latest version npm version: Latest version 

If anyone has encountered a similar issue or possesses knowledge in this area, your insights would be greatly appreciated. Specifically, I am looking for: 
- Advanced troubleshooting techniques that might be applicable in this scenario? 
- Insights into potential causes that are not typically addressed by standard procedures? 
- Any additional steps or configurations that I may have overlooked? 

FYI: I went through the related questions list in "Stack Overflow", but I couldn't resolve the issue: 
- https://stackoverflow.com/questions/60379393/cypress-installation-fails-with-proxy-setup
- https://stackoverflow.com/questions/36494336/npm-install-error-unable-to-get-local-issuer-certificate

Note: In this question, I have altered specific network details such as domain names, proxy addresses, and port numbers for security and privacy reasons. Please consider these as placeholders while providing any advice or solutions. Thank you for understanding. 

**Set Proxy Configuration**: 
- `npm config set proxy http://proxy.example.net:70 `
- `npm config set https-proxy http://proxy.example.net:70 `

**For Setting Environment Variables**: 
- `set HTTPS_PROXY=http://domain%5Cusername:password@proxy.example.net:70 `
- `set HTTP_PROXY=http://domain%5Cusername:password@proxy.example.net:70`
- `set HTTPS_PROXY=https://proxy.example.net:70` 
- `set http-proxy=http://proxy.example.net:70` 
- `set SOME_VARIABLE=some-value`

**For npm Configuration**:
- `npm config set proxy http://proxy.example.net:70`
- `npm config set http-proxy http://proxy.example.net:70` 

**For Git configuration**:
- `git config --global https.proxy http://proxy.example.net:70` 
- `git config --global http.proxy http://proxy.example.net:70` 

**For Certificate Issues**:(Consult the security team before executing these commands) 
- `npm set strict-ssl=false` 
- `set NODE_TLS_REJECT_UNAUTHORIZED=0` 

**Installing Cypress locally**: 
I followed the installation instructions provided by Cypress, installing it as a desktop application on my local machine, After installation, I attempted to run Cypress, but I faced issues possibly related to my corporate proxy settings. 

**Seek help**: 
- Dive deep into cypress documentation. 
- Reach out to our Corp-Networking team None of the above resolved the SSL/TLS certificate issue!
 
CC: @cypress-github-team-services-sa When I configured the environment & npm & HTTP/HTTPS & Git and the Certificate there were no indications of errors within the CMD.

I executed `npm install cypress --save-dev` again -> Same error showed up again -> `npm ERR! code ERR_SSL_TLSV1_UNRECOGNIZE`
Can anyone help me overcome this issue, please?

**Comment by portyu9** [comment]
Hi @KSHCR7 

The important part of this error is:

```text
ERR_SSL_TLSV1_UNRECOGNIZED_NAME
tlsv1 unrecognized name
```

That is different from a normal certificate-trust error such as `unable to verify the first certificate`.

It strongly suggests the TLS endpoint/proxy is rejecting the hostname/SNI being presented. Because of that, settings such as:

```text
strict-ssl=false
NODE_TLS_REJECT_UNAUTHORIZED=0
```

are not expected to fix the underlying problem.

I would first get the **exact canonical proxy URL, protocol, and port** from your network team. A proxy used for HTTPS destinations is very often still configured with an HTTP proxy URL because HTTPS is tunneled through HTTP `CONNECT`, for example:

```text
HTTPS_PROXY=http://proxy.example.net:8080
```

rather than:

```text
HTTPS_PROXY=https://proxy.example.net:8080
```

but you should use whatever your corporate proxy actually supports.

Also remove conflicting proxy definitions while debugging. Cypress gives environment variables priority, so an old `HTTPS_PROXY` can override the npm settings you are changing.

Run:

```bash
npx cypress info
```

to confirm which proxy Cypress sees.

If your company performs TLS interception with its own certificate authority, configure that CA through npm for the Cypress **binary installation**, for example:

```bash
npm config set cafile "C:\path\to\corporate-ca.pem"
```

Cypress's current documentation specifically notes that `NODE_EXTRA_CA_CERTS` is not used by the install-time binary download; npm `cafile`/`ca` is the supported trust source there.

Git's proxy configuration is unrelated to the Cypress binary download, so changing Git's global proxy will not solve this particular installation path.

Given the `unrecognized_name` alert, I would have the network team verify the proxy's expected SNI/CONNECT behavior first rather than disabling TLS validation.

Please mark this answer as accepted if it helped, thank you.
